src/App.tsx

```diff
@@ -8,7 +8,46 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { FILTER_STATUS } from './utils/FILTER_STATUS';
+import { getActive } from './services/todo';
+import { getCompleted } from './services/todo';
+import { getAllTodos } from './services/todo';
+import { handleQuery } from './utils/handleQuery';
+
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [filterBy, setFilterBy] = useState(FILTER_STATUS.ALL);
+  const [query, setQuery] = useState('');
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    switch (filterBy) {
+      case FILTER_STATUS.ALL:
+        setIsLoading(true);
+
+        getAllTodos()
+          .then(result => setTodos(handleQuery(result, query)))
+          .finally(() => setIsLoading(false));
+        break;
+      case FILTER_STATUS.ACTIVE:
+        setIsLoading(true);
+        getActive()
+          .then(result => setTodos(handleQuery(result, query)))
+          .finally(() => setIsLoading(false));
+        break;
+      case FILTER_STATUS.COMPLETED:
+        setIsLoading(true);
+
+        getCompleted()
+          .then(result => setTodos(handleQuery(result, query)))
+          .finally(() => setIsLoading(false));
+        break;
+    }
+  }, [filterBy, query]);
```

Comment:
You should make one get request here with an empty array of dependencies because in this way you did you make it each time you change state of app
```suggestion
  useEffect(() => {
    //set is loading true
    //make one request here 
    //set isloading false
    }
  }, []);
```